processing/src/main/java/io/druid/query/filter/SelectorDimFilter.java

```diff
@@ -96,27 +144,22 @@ public boolean equals(Object o)
 
     SelectorDimFilter that = (SelectorDimFilter) o;
 
-    if (dimension != null ? !dimension.equals(that.dimension) : that.dimension != null) {
+    if (!dimension.equals(that.dimension)) {
       return false;
     }
     if (value != null ? !value.equals(that.value) : that.value != null) {
       return false;
     }
+    return extractionFn != null ? extractionFn.equals(that.extractionFn) : that.extractionFn == null;
```

Comment:
Do all the extraction functions have proper equals implementations? I haven't actually checked.